File: demos/android/demos/src/main/java/com/antgroup/antv/f2/kotlinsamples/charts/SingleAreaChart_1.java
package com.antgroup.antv.f2.kotlinsamples.charts;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import com.antgroup.antv.f2.F2CanvasView;
import com.antgroup.antv.f2.F2Chart;
import com.antgroup.antv.f2.F2Config;
import com.antgroup.antv.f2.F2Function;
import com.antgroup.antv.f2.F2Util;
import com.antgroup.antv.f2.kotlinsamples.Utils;

/**
 * @author qingyuan.yl
 * @date 2020-09-27
 */
public class SingleAreaChart_1 implements F2CanvasView.Adapter {

    private F2Chart mChart;

    @Override
    public void onCanvasDraw(F2CanvasView canvasView) {
        if (mChart == null) {
            mChart = F2Chart.create(canvasView.getContext(), "SingleAreaChart_1", canvasView.getWidth(), canvasView.getHeight());
        }

        mChart.setCanvas(canvasView);
        mChart.padding(10, 0, 0, 0);
        mChart.source(Utils.loadAssetFile(canvasView.getContext(), "mockData_singleAreaChart.json"));

        mChart.setAxis("time", new F2Chart.AxisConfigBuilder()
                .label(new F2Chart.AxisLabelConfigBuilder().labelOffset(5.f))
                .gridHidden());

        mChart.setAxis("tem", new F2Chart.AxisConfigBuilder()
                .grid(new F2Chart.AxisGridConfigBuilder().type("dash")));
        mChart.setScale("tem", new F2Chart.ScaleConfigBuilder().min(800).max(3000));

        mChart.line().position("time*tem");
        mChart.area().position("time*tem").fixedColor(
                new F2Util.ColorLinearGradient()
                        .addColorStop(0.f, "red")
                        .addColorStop(1.f, "white")
                        .setPosition(0, 0, 0, canvasView.getHeight())
        );
        mChart.point().position("time*tem").fixedColor("#FFB6C1").style(new F2Config.Builder()
                .setOption("custom", mChart, new F2Function() {
                    @Override
                    public F2Config execute(String param) {
                        JSONObject jsonObject = JSON.parseObject(param);
                        if (jsonObject.getIntValue("_index") % 2 == 0) {
                            return new F2Config.Builder().setOption("size", 0).build();
                        }
                        return null;
                    }
                })
                .build());

        mChart.render();
    }

    @Override
    public void onDestroy() {
        if (mChart != null) {
            mChart.destroy();
        }
    }
}

File: demos/android/demos-kotlin/src/main/java/com/antgroup/antv/f2/kotlinsamples/charts/SingleAreaChart_1.kt
package com.antgroup.antv.f2.kotlinsamples.charts

import com.alibaba.fastjson.JSON
import com.antgroup.antv.f2.F2CanvasView
import com.antgroup.antv.f2.F2Chart
import com.antgroup.antv.f2.F2Chart.*
import com.antgroup.antv.f2.F2Config
import com.antgroup.antv.f2.F2Function
import com.antgroup.antv.f2.F2Util.*
import com.antgroup.antv.f2.kotlinsamples.Utils.loadAssetFile

/**
 * @author qingyuan.yl
 * @date 2020-09-27
 */
class SingleAreaChart_1 : F2CanvasView.Adapter {
    private var mChart: F2Chart? = null
    override fun onCanvasDraw(canvasView: F2CanvasView) {
        if (mChart == null) {
            mChart = F2Chart.create(
                canvasView.context,
                "SingleAreaChart_1",
                canvasView.width.toDouble(),
                canvasView.height.toDouble()
            )
        }
        mChart!!.setCanvas(canvasView)
        mChart!!.padding(10.0, 0.0, 0.0, 0.0)
        mChart!!.source(loadAssetFile(canvasView.context, "mockData_singleAreaChart.json"))
        mChart!!.setAxis(
            "time", AxisConfigBuilder()
                .label(AxisLabelConfigBuilder().labelOffset(5f))
                .gridHidden()
        )
        mChart!!.setAxis(
            "tem", AxisConfigBuilder()
                .grid(AxisGridConfigBuilder().type("dash"))
        )
        mChart!!.setScale("tem", ScaleConfigBuilder().min(800.0).max(3000.0))
        mChart!!.line().position("time*tem")
        mChart!!.area().position("time*tem").fixedColor(
            ColorLinearGradient()
                .addColorStop(0f, "red")
                .addColorStop(1f, "white")
                .setPosition(0f, 0f, 0f, canvasView.height.toFloat())
        )
        mChart!!.point().position("time*tem").fixedColor("#FFB6C1").style(
            F2Config.Builder<F2Config.Builder<*>>()
                .setOption("custom", mChart, object : F2Function() {
                    override fun execute(param: String): F2Config {
                        val jsonObject = JSON.parseObject(param)
                        return if (jsonObject.getIntValue("_index") % 2 == 0) {
                            F2Config.Builder<F2Config.Builder<*>>().setOption("size", 0).build()
                        } else F2Config.Builder<F2Config.Builder<*>>().build()
                    }
                })
                .build()
        )
        mChart!!.render()
    }

    override fun onDestroy() {
        if (mChart != null) {
            mChart!!.destroy()
        }
    }
}